rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -77,17 +77,8 @@ public T doExecute(Object... args) throws Throwable {
                     if (StringUtils.isNullOrEmpty(lockKeys)) {
                         break;
                     }
-
-                    if (RootContext.inGlobalTransaction()) {
-                        //do as usual
-                        statementProxy.getConnectionProxy().checkLock(lockKeys);
-                    } else if (RootContext.requireGlobalLock()) {
-                        //check lock key before commit just like DML to avoid reentrant lock problem(no xid thus can
-                        // not reentrant)
-                        statementProxy.getConnectionProxy().appendLockKey(lockKeys);
-                    } else {
-                        throw new RuntimeException("Unknown situation!");
-                    }
+                    // check pk lock to TC
+                    statementProxy.getConnectionProxy().checkLock(lockKeys);
```

Comment:
Regardless of the append or check here, there is still a problem that the isolation cannot be guaranteed between the select and the subsequent business. Are you considering adding locks instead of checking locks?

Reply:
I think if user want to run a transaction in READ_COMMITED level, he could use @GlobalLock and `select for update` together. Normal use case maybe like follow:
```java
@Trancation
@GlobalLock
public void localTransaction() {
      // select pk from table where pk = 1 for update 
      // update table set column = value where pk = 1
      // commit after method finished
}
```
In this local transaction, it get the DB's row lock, then seata will check TC's lock, if no global transaction use pk = 1, then this local transaction could invoke the update DML. I think check lock to TC is enough, because DB's row lock will been holded by this local transaction, so nobody could get DB's row lock anymore. After finish local transaction, it release the DB's row lock.

Reply:
what if selectFor update and update in two rpc call ?